src/app/ReadHandler.cpp

```diff
@@ -652,9 +716,59 @@ CHIP_ERROR ReadHandler::ProcessSubscribeRequest(System::PacketBufferHandle && aP
 
     mExchangeCtx->WillSendMessage();
 
+#if CHIP_CONFIG_PERSIST_SUBSCRIPTIONS
+    PersistSubscription();
+#endif // CHIP_CONFIG_PERSIST_SUBSCRIPTIONS
+
     return CHIP_NO_ERROR;
 }
 
+void ReadHandler::PersistSubscription()
+{
+    auto * subscriptionResumptionStorage = InteractionModelEngine::GetInstance()->GetSubscriptionResumptionStorage();
+    VerifyOrReturn(subscriptionResumptionStorage != nullptr);
+
+    SubscriptionResumptionStorage::SubscriptionInfo subscriptionInfo = { .mNodeId         = GetInitiatorNodeId(),
+                                                                         .mFabricIndex    = GetAccessingFabricIndex(),
+                                                                         .mSubscriptionId = mSubscriptionId,
+                                                                         .mMinInterval    = mMinIntervalFloorSeconds,
+                                                                         .mMaxInterval    = mMaxInterval,
+                                                                         .mFabricFiltered = IsFabricFiltered() };
+    ObjectList<AttributePathParams> * attributePath                  = mpAttributePathList;
+    size_t attributePathCount                                        = 0;
+    while (attributePath)
+    {
+        attributePathCount++;
+        attributePath = attributePath->mpNext;
+    }
+    attributePath = mpAttributePathList;
+    subscriptionInfo.mAttributePaths.Calloc(attributePathCount);
```

Comment:
It might be sensible to move this copy code to methods on `SubscriptionResumptionStorage::SubscriptionInfo` taking arguments `ObjectList<AttributePathParams>` and `ObjectList<EventPathParams>` so that `ReadHandler` doesn't need to know how to do this.

That would also solve this problem: this feature assumes that `AllocatedSize()` can fit within `UINT16_MAX`:

https://github.com/project-chip/connectedhomeip/blob/866ba53802cfa723a23c4c89a09ccf230080d198/src/app/SimpleSubscriptionResumptionStorage.cpp#L251

That's sensible because it is impossible for our storage to hold something larger than that.  But if `SubscriptionResumptionStorage::SubscriptionInfo` was doing the copying, it could enforce this, whereas `ReadHandler` isn't in a position to do so.

Reply:
Okay that's sensible. I've added `SetAttributePaths` and `SetEventPaths` to `SubscriptionInfo` and have ReadHandler call those with the respective ObjectLists.